src/gui/issueswidget.cpp

```diff
@@ -196,6 +196,17 @@ void IssuesWidget::addItem(QTreeWidgetItem *item)
         }
     }
 
+    // Wipe any existing message for the same folder and path
+    auto newData = ProtocolItem::extraData(item);
+    for (int i = 0; i < count; ++i) {
```

Comment:
hopefully this does not ruins the performence when adding many items.
This looks like O(n²) complexity. anything we can do about that?

Maybe storing a QSet of existing items so we only do that for duplicate items?

Reply:
Yes, we've had performance problems here before after all. I think this should be fine though since we're log n faster than last time and also limit n to 50k. 

Nevertheless, I had rejected adding a set because I thought it'd be too costly memory wise - but was just a mistake (as long as we don't copy any string data). Then the only remaining drawback is that adding a set means keeping the two in sync which shouldn't be an obstacle.